includes/rest-api/mappers/class-sensei-rest-api-home-controller-mapper.php

```diff
@@ -112,4 +112,55 @@ public function map_promo_banner( Sensei_Home_Promo_Banner $banner ): array {
 			'is_visible' => $banner->is_visible(),
 		];
 	}
+
+	/**
+	 * Maps a Sensei_Home_Tasks to a basic array structure to be used as response for the REST API.
+	 *
+	 * @param Sensei_Home_Tasks $tasks The tasks structure.
+	 *
+	 * @return array
+	 */
+	public function map_tasks( Sensei_Home_Tasks $tasks ): array {
+		$mapped = [];
+		foreach ( $tasks->get_items() as $task ) {
+			$mapped[ $task->get_id() ] = $this->map_task( $task );
+		}
+
+		return [
+			/**
+			 * Filter to alter the list of tasks that will be sent to the frontend.
+			 * New tasks can be added and existing tasks modified or removed.
+			 * Each item must have:
+			 * - title: the title of the task.
+			 * - priority: number used in frontend to sort tasks.
+			 * - url: optional string with a url users will be taken when clicking on the task.
+			 * - image: optional string with a path/url to a featured image.
+			 * - done: boolean to mark task as done or not.
+			 *
+			 * @hook sensei_home_tasks_items
+			 * @since $$next-version$$
+			 *
+			 * @param array $tasks An array indexed by task ID.
+			 *
+			 * @return {bool}
+			 */
+			'items' => apply_filters( 'sensei_home_tasks_items', $mapped ),
```

Comment:
WDYT about maybe applying this filter directly on `$tasks`, and adding a new method like `add_task` on `Sensei_Home_Tasks`, so we can add tasks properly and have them properly formatted here automatically for us?

Reply:
I am glad you raised this, because I don't like the result and maybe I did some wrong assumption:
I did this in this way because I didn't want to expose `Sensei_Home_Tasks` (nor `Sensei_Home_Task`) since I am assuming that other plugin developers might not have access to those classes.

So that's why I am applying the filter AFTER mapping to a plain array structure. What do you think on this? Is it safe to assume that they will have access to any of those classes?

Reply:
Hey there. Very good point indeed!

WDYT about maybe having a class that implements `Sensei_Home_Task`, but returns the data from an array instead, similar to the structure that this filter needs to provide?

I'm asking because at least, in this case, we would be able to provide an `add_task` method on `Sensei_Home_Tasks`, and then we would have some validation on the array format, which is good when working with these filters.

In the same way, we would be able to provide a `remove_task` method, to remove a given task by ID, too, which would be nice.

WDYT?

Reply:
I am sorry Fernando, but I am not sure I am following you correctly. I will open an slack with you so we can talk about this today :)